Mathlib/RingTheory/MvPowerSeries/Rename.lean

```diff
@@ -0,0 +1,160 @@
+/-
+Copyright (c) 2025 Riccardo Brasca. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Riccardo Brasca
+-/
+import Mathlib.RingTheory.MvPowerSeries.Substitution
+
+/-!
+# Renaming variables of power series.
+
+This file establishes the `rename` operation on multivariate power series,
+which modifies the set of variables. Note that the function `f : σ → τ` must satisfies
+`Tendsto f cofinite cofinite`, otherwise renaming does not make sense. This is the case if `f`
+is injective (`Injective.tendsto_cofinite`) or if `σ` is a `Fintype` (`simp` can prove this).
+
+## Main declarations
+
+* `MvPowerSeries.rename`
+
+## Notation
+
+As in other files, we typically use the notation:
+
++ `σ τ α : Type*` (indexing the variables)
+
++ `R S : Type*` `[CommSemiring R]` `[CommSemiring S]` (the coefficients)
+
++ `s : σ →₀ ℕ`, a function from `σ` to `ℕ` which is zero away from a finite set.
+This will give rise to a monomial in `MvPowerSeries σ R` which mathematicians might call `X^s`
+
++ `r : R` elements of the coefficient ring
+
++ `i : σ`, with corresponding monomial `X i`, often denoted `X_i` by mathematicians
+
++ `F : MvPowerSeries σ α`
+
+## TODO
+Finish to copy the file `RingTheory.MvPolynomial.rename`.
+
+-/
+
+noncomputable section
+
+variable {R S σ τ α : Type*}
+
+namespace MvPowerSeries
+
+section Rename
+
+open Function Filter
+
+variable [CommRing R] {f : σ → τ} (hf : Tendsto f cofinite cofinite)
+
+include hf
+
+/-- Rename all the variables in a multivariable power series. -/
+def rename : MvPowerSeries σ R →ₐ[R] MvPowerSeries τ R :=
+  substAlgHom (hasSubst_X_comp hf)
+
+theorem rename_C (r : R) : rename hf (C σ R r) = C τ R r := by
+  simp [rename, c_eq_algebraMap]
+
+@[simp]
+theorem rename_X (i : σ) : rename hf (X i : MvPowerSeries σ R) = X (f i) := by
+  simp [rename, hasSubst_X_comp hf]
+
+@[simp]
+theorem rename_monomial (n : σ →₀ ℕ) (r : R) :
```

Comment:
I believe you wish to have a `rename_coe` that works for polynomials, not only monomials.